### ISSUE TITLE
Manuform: Gapless 7-key thumb cluster

### ISSUE THREAD

**Issue by jgandert** [ibnuda/dactyl-keyboard#32]
Having looked a bit into the source code (without Clojure knowledge), the thumb cluster of the Manuform is more hard-coded than the Lightcycle, which seems to make #28 very time consuming to implement. 

Additionally, changing the curvature of the thumb cluster doesn't fully address a problem that I feared and have now seen in a lot of reviews / comments (e.g. [1](https://www.reddit.com/r/ErgoMechKeyboards/comments/d89rkz/help_want_to_build_dactylmanuform_but_which_is/f196zc7/),  [2](https://www.reddit.com/r/MechanicalKeyboards/comments/ajgdg5/criticisms_of_dactylmanuform_dacman_from_another/), [3](https://www.reddit.com/r/MechanicalKeyboards/comments/9d0gqn/just_finished_my_first_dactyl_manuform_i_cant/e5ey4d4/), [4](https://www.reddit.com/r/ErgoMechKeyboards/comments/hfug5k/how_to_best_hit_the_middle_thumb_key_of_a_dactyl/fw1bi1g/)):  the outer thumb keys being hard to reach.

As far as I can see, this is partially a result of the downward curvature (instead of upward) and the horizontal rotation of keys, which requires gaps and increased distance so that keycaps don't touch.

So, it seems that something like the following 7-key layout would be really useful to have:

![Screenshot_2020-06-28 Keyboard Layout Editor cut 2](https://user-images.githubusercontent.com/3063858/85959141-e1a4d680-b99a-11ea-8067-e27598569a80.png)

The ones marked with `(tented)` would be [tented](https://user-images.githubusercontent.com/3063858/85959583-9dff9c00-b99d-11ea-848f-791090969b77.png) toward the center of the two leftmost keys, to make them easier to reach. It would look kind of like this when looking down from above the **1** key:

![drawing svg](https://user-images.githubusercontent.com/3063858/85959148-e9fd1180-b99a-11ea-8cb4-4043fc554335.png)

(the outer column would be tented the same amount towards the middle column, as the middle column is tented towards the leftmost column)

The reason for the shift in the last column is that this keeps much of the optimal curve (marked in green here) from Minidox and other Manuform thumb clusters:

![untitled3 curve](https://user-images.githubusercontent.com/3063858/85958589-da7bc980-b996-11ea-8ba8-b4e5a23883e9.png)

And here's the layout added on top of the Manuform layout (just for fun):

![untitled2](https://user-images.githubusercontent.com/3063858/85958414-b9ff3f80-b995-11ea-837e-74e6293d6ae9.png)

**Comment by dereknheiley** [comment]
my next version will have the mini 5-thumb cluster, since i also find the stock dactyl-manuform thumb buttons to not be in the best position, that said, the dactyl, and similar kinesis and ergo-dox thumb clusters are pretty busy with that many keys in close proximity, have you tried simulating this with a key switch tester or some creative hot glue-ing?

**Comment by jgandert** [comment]
Haven't tested it yet but I use the Kinesis Advantage atm. which has a 6-key thumb cluster with relative close proximity between keys and I actually find the outer keys (alt, home, end) too far away. 
 
(Of course this depends on your hand. I think mine are a bit larger than average.)

![Untitled](https://user-images.githubusercontent.com/3063858/87063127-3acc0000-c20e-11ea-81f8-164c484031ba.png)

Another thing I've noticed with the Advantage is that I almost only hit the two 2u keys at the top, so given the right angle, having another two keys would be nice. That said, the lower half of the Advantage spacebar key might not be that comfortable to hit, so a 6-key thumb cluster without the bottom 1u key might be totally fine and maybe even better.

![Screenshot_2020-07-09 Keyboard Layout Editor 2](https://user-images.githubusercontent.com/3063858/87087898-31558e80-c234-11ea-8a87-1a73d52d7a4e.png)



Also from having read and watched a lot of reviews of both the Advantage and the Dactyl (Manuform and not), I'd say that the thumb cluster seems to be the area of those keyboards that has the biggest potential for improvement. A lot of reviews mention they're not quite happy with the thumb cluster. 

The Dactyl also has a downwards slope for the thumb cluster, and some reviews (that I agree with) mentioned that it doesn't seem to make sense that the thumb keys are curving away from your hand, rather than curving towards you as is the case for the main letter keys.

One keyboard that does that is the [Tightyl](https://github.com/okke-formsma/dactyl-manuform-tight) by @okke-formsma (also based on the Dactyl by @adereth):

![tightyl](https://user-images.githubusercontent.com/3063858/87067875-014ac300-c215-11ea-88e9-8672ee92f983.png)


*(image source: https://imgur.com/a/YePWDY5#HrvDgbh)*

**Comment by dereknheiley** [comment]
yeah, it would be nice to try a / have the ability to adjust the convex/concave param for, a concave version of, the "5 (Mini)"
